lambda/update/correction.go

```diff
@@ -36,6 +36,13 @@ type DonorCorrection struct {
 }
 
 func (c DonorCorrection) Apply(lpa *shared.Lpa) []shared.FieldError {
+	if lpa.Donor.IdentityCheck != nil && !c.DateOfBirth.IsZero() && c.DateOfBirth != lpa.Donor.DateOfBirth {
```

Comment:
Maybe pull this condition into a var just above as its not the most straightforward one to understand 
Something like donorIdCheckWithADob